src/backend/cdb/cdbmutate.c

```diff
@@ -845,7 +845,8 @@ apply_motion_mutator(Node *node, ApplyMotionState *context)
 	if (IsA(newnode, Motion) &&flow->req_move != MOVEMENT_NONE)
 	{
 		plan = ((Motion *) newnode)->plan.lefttree;
-		flow = plan->flow;
+		flow->flow_before_req_move = plan->flow;
+		plan->flow = flow;
```

Comment:
This change must be described in a comment. Also, you should explain in PR description why this code block affect your patch and what is the problem here.

Reply:
> This change must be described in a comment.

done